full-stack-tests/backend/src/integration/Checkpoints.test.ts

```diff
@@ -153,36 +152,26 @@ describe("Checkpoints", () => {
     });
     const prefetchSpy = sinon.spy(CloudSqlite, "startCloudPrefetch").withArgs(sinon.match.any, `${testChangeSet.id}.bim`, sinon.match.any); // Need matchers because GCS is also prefetched.
     const settingsSpy = sinon.spy(IModelHost.appWorkspace.settings, "getBoolean").withArgs("Checkpoints/prefetch");
-    const iModel = await SnapshotDb.openCheckpointV2({
+    const iModel = await SnapshotDb.openCheckpointFromRpc({
       accessToken,
       iTwinId: testITwinId,
       iModelId: testIModelId,
       changeset: testChangeSet,
     });
     expect(prefetchSpy.callCount).to.equal(1);
     expect(settingsSpy.callCount).to.equal(1);
-    // stub openFile the second time around, we only need the attach call from openCheckpointV2 to take place to trigger a prefetch.
-    sinon.stub(SnapshotDb, "openFile").callsFake(() => {
-      return {} as SnapshotDb;
-    });
-    sinon.stub(CheckpointManager, "validateCheckpointGuids").callsFake(() => { });
-
-    await SnapshotDb.openCheckpointV2({
-      accessToken,
-      iTwinId: testITwinId,
-      iModelId: testIModelId,
-      changeset: testChangeSet,
-    });
-    expect(prefetchSpy.callCount).to.equal(1);
```

Comment:
A test which makes sure we don't double prefetch seems valuable to me. Do we need to remove this? 
This confirms that two backends which open the same iModels don't trigger a prefetch more than once ( thus giving that prefetch more weight ).

Reply:
It is illegal to open the same iModel more than once from a backend. You stubbed out "openFile" and "validate.." to avoid that rule, so this "issue" really only applies to daemons. We never open the same iModel/changeset from two backends on the same vm. Even if we did, prefetching an already perfectched block doesn't do anything.

Reply:
You're right this issue only applies to daemons, but why shouldn't we test it (even if its a simulated version of a test where I had to stub some calls out?).

>We never open the same iModel/changeset from two backends on the same vm

DR has started deferring rpc calls to certain interfaces(tile, read, presentation) to GPB. So that would mean two backends reading the same imodel/changeset. Prefetching an already prefetched block does do nothing, you're right. But there will be two prefetch objects prefetching the same iModel if we were to allow two prefetches on the same iModel.

Reply:
i re-added that test, but i'm not really even sure about the policy. If we really do have two backends on the same VM sharing a daemon and they both are using the same checkpoint, why *not* start two prefetches. I'm not sure two prefetch requests will actually do anything different than one, but certainly if they did somehow bump the priority for that checkpoint over others, isn't that a good thing (given there are two clients waiting for it)?

Reply:
From my limited testing, it did do something different in that more prefetches occurred for that checkpoint. You have a good point, but in this particular scenario I guess I view this as an artificial inflation of 1 client to 2 clients, since GPB is really only there to serve the DR backend's requests and not because someone else opened the same checkpoint in some other frontend application.